cmd/terramate/cli/cloud_sync_terraform_plan.go

```diff
@@ -33,10 +33,13 @@ func (c *cli) getTerraformChangeset(run stackCloudRun, planfile string) (*cloud.
 		return nil, errors.E(clitest.ErrCloudInvalidTerraformPlanFilePath, "path must be relative to the running stack")
 	}
 
-	absPlanFilePath := filepath.Join(run.Stack.HostDir(c.cfg()), planfile)
-	_, err := os.Lstat(absPlanFilePath)
-	if err != nil {
-		return nil, errors.E(err, "checking plan file")
+	// Terragrunt writes the plan to a temporary directory, so we cannot check for its existence.
+	if !run.Task.UseTerragrunt {
+		absPlanFilePath := filepath.Join(run.Stack.HostDir(c.cfg()), planfile)
+		_, err := os.Lstat(absPlanFilePath)
+		if err != nil {
+			return nil, errors.E(err, "checking plan file")
+		}
```

Comment:
what about removing this? this special handling for terraform maybe can create more issues in the future when we introduce more integrations.